pkg/providers/property/data_akamai_search_test.go

```diff
@@ -0,0 +1,70 @@
+package property
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/stretchr/testify/mock"
+
+	"github.com/akamai/AkamaiOPEN-edgegrid-golang/v2/pkg/papi"
+)
+
+func TestDSSearch(t *testing.T) {
+	t.Run("match by hostname", func(t *testing.T) {
+		client := &mockpapi{}
+
+		// hosntame provided by fixture is "www.akamai.com"
+		search := papi.SearchItems{Items: []papi.SearchItem{
+			{
+				AccountID:        "acc_test",
+				AssetID:          "aid_test",
+				GroupID:          "grp_test",
+				ContractID:       "ctr_test",
+				EdgeHostname:     "www.akamai.com.edgekey.net",
+				Hostname:         "www.akamai.com",
+				ProductionStatus: "INACTIVE",
+				StagingStatus:    "ACTIVE",
+				PropertyID:       "prp_test",
+				PropertyName:     "test_www.akamai.com",
+				PropertyVersion:  1,
+				UpdatedByUser:    "test_user@akamai.com",
+				UpdatedDate:      "2021-11-22T07:24:56Z",
+			},
+			{
+				AccountID:        "acc_test",
+				AssetID:          "aid_test",
+				GroupID:          "grp_test",
+				ContractID:       "ctr_test",
+				EdgeHostname:     "www.akamai.com.edgekey.net",
+				Hostname:         "www.akamai.com",
+				ProductionStatus: "ACTIVE",
+				StagingStatus:    "INACTIVE",
+				PropertyID:       "prp_test1",
+				PropertyName:     "test1_www.akamai.com",
+				PropertyVersion:  1,
+				UpdatedByUser:    "test_user@akamai.com",
+				UpdatedDate:      "2021-11-22T07:24:56Z",
+			},
+		}}
+
+		client.On("SearchProperties",
+			mock.Anything, // ctx is irrelevant for this test
+			papi.SearchRequest{Key: "hostname", Value: "www.akamai.com"},
+		).Return(&papi.SearchResponse{Versions: search}, nil)
+
+		useClient(client, func() {
+			resource.UnitTest(t, resource.TestCase{
+				Providers: testAccProviders,
+				Steps: []resource.TestStep{{
+					Config: loadFixtureString("testdata/TestDSSearch/match_by_hostname.tf"),
+					Check: resource.ComposeAggregateTestCheckFunc(
+						resource.TestCheckResourceAttrSet("data.akamai_search.test", "id"),
```

Comment:
```suggestion
						resource.TestCheckResourceAttr("data.akamai_search.test", "id", "hostname:www.example.com"),
```

Reply:
obviously better www.akamai.com, given your test tf

Reply:
changed and used www.example.com but moved to pkg/providers/property/data_akamai_property_search_test.go